ostd/src/arch/x86/mm/mod.rs

```diff
@@ -270,21 +308,60 @@ impl PageTableEntryTrait for PageTableEntry {
                 PageTableFlags::SHARED
             );
         }
+
+        // Set cache policy using PAT, PCD, PWT bits
+        // For 4KB pages (`VALID_PAGE` set), bit 7 is PAT
+        // For huge pages (`HUGE` set), bit 7 is `HUGE` flag, cannot use PAT
+        // Preserve the `HUGE` and `VALID_PAGE` bits to determine page type
+        let is_4kb_page = (self.0 & PageTableFlags::VALID_PAGE.bits()) != 0;
+        let is_huge_page = (self.0 & PageTableFlags::HUGE.bits()) != 0;
+
+        // Preserve `HUGE` or `VALID_PAGE` bit in flags
+        if is_4kb_page {
+            flags |= PageTableFlags::VALID_PAGE.bits();
+        } else if is_huge_page {
+            flags |= PageTableFlags::HUGE.bits();
+        }
+
         match prop.cache {
-            CachePolicy::Writeback => {}
+            CachePolicy::Writeback => {
+                // PAT=0, PCD=0, PWT=0 (default, no bits set)
+            }
             CachePolicy::Writethrough => {
+                // PAT=0, PCD=0, PWT=1
                 flags |= PageTableFlags::WRITE_THROUGH.bits();
             }
             CachePolicy::Uncacheable => {
+                // PAT=0, PCD=1, PWT=0
                 flags |= PageTableFlags::NO_CACHE.bits();
             }
-            _ => panic!("unsupported cache policy"),
+            CachePolicy::WriteCombining => {
+                // PAT=1, PCD=0, PWT=0
+                // Only valid for 4KB pages; for 4KB pages, set bit 7 as PAT
+                if is_4kb_page {
+                    flags |= PageTableFlags::HUGE.bits(); // This is PAT bit for level 1
+                } else {
+                    panic!("WriteCombining only supported for 4KB pages, not huge pages");
+                }
+            }
+            CachePolicy::WriteProtected => {
+                // PAT=1, PCD=0, PWT=1
+                // Only valid for 4KB pages
+                if is_4kb_page {
+                    flags |= PageTableFlags::HUGE.bits() | PageTableFlags::WRITE_THROUGH.bits();
+                } else {
+                    panic!("WriteProtected only supported for 4KB pages, not huge pages");
+                }
+            }
         }
         self.0 = self.0 & !Self::PROP_MASK | flags;
     }
 
     fn is_last(&self, _level: PagingLevel) -> bool {
-        self.0 & PageTableFlags::HUGE.bits() != 0
+        // A PTE is "last" (leaf mapping) if:
+        // 1. It has HUGE bit set (huge page at level 2/3), OR
+        // 2. It has VALID_PAGE bit set (4KB page at level 1)
+        self.0 & PageTableFlags::HUGE.bits() != 0 || self.0 & PageTableFlags::VALID_PAGE.bits() != 0
```

Comment:
We do we want to do this instead of checking `level == 1 || self.0.contains(PageTableFlags::HUGE)`?

https://github.com/asterinas/asterinas/blob/0615e6f6424ead8e12d73670428149fa7270571f/ostd/src/arch/loongarch/mm/mod.rs#L289-L291

This can simplify the related logics a lot.